lib/internal/streams/readable.js

```diff
@@ -1076,14 +1085,19 @@ Readable.prototype[SymbolAsyncIterator] = function() {
     }).wrap(src);
   }
 
-  const iter = createAsyncIterator(stream);
+  const iter = createAsyncIterator(stream, options);
   iter.stream = stream;
   return iter;
-};
+}
 
-async function* createAsyncIterator(stream) {
+async function* createAsyncIterator(stream, options) {
   let callback = nop;
 
+  const {
+    destroyOnReturn = true,
+    destroyOnError = true,
+  } = (options ?? {});
```

Comment:
Can you please use an `Object.assign()` instead?

Reply:
used { ...options} as that's what the linter demanded.